### PR TITLE
[Performance] Optimize unmarshalling

### PR DESCRIPTION
**Pull request by dirkkul**
### What's being changed:

Smarter unmarshalling from binary

### CODE REVIEW

File: entities/storobj/storage_object.go [hunk 1]
```diff
@@ -113,94 +111,81 @@ func FromBinaryUUIDOnly(data []byte) (*Object, error) {
...
-	ec.AddWrap(binary.Read(r, le, &kindByte), "kind")
-	_, err := r.Read(uuidBytes)
-	ec.AddWrap(err, "uuid")
-	ec.AddWrap(binary.Read(r, le, &createTime), "create time")
-	ec.AddWrap(binary.Read(r, le, &updateTime), "update time")
-	ec.AddWrap(binary.Read(r, le, &vectorLength), "vector length")
-	ko.VectorLen = int(vectorLength)
+	createTime := int64(rw.ReadUint64())
+	updateTime := int64(rw.ReadUint64())
+	vectorLength2 := rw.ReadUint16()
```
Comment (parkerduckworth) [inline entities/storobj/storage_object.go:131]: > `vectorLength2`

Does the `2` have significance here?

File: entities/storobj/storage_object.go [hunk 2]
```diff
@@ -113,94 +111,81 @@ func FromBinaryUUIDOnly(data []byte) (*Object, error) {
...
+		rw.MoveBufferPositionForward(uint64(metaLength))
+	}
+
+	vectorWeightsLength := rw.ReadUint32()
+	vectorWeights := rw.ReadBytesFromBuffer(uint64(vectorWeightsLength))
+
+	// some object members need additional "enrichment". Only do this if necessary
+	if len(props) > 0 ||
+		len(meta) > 0 ||
+		vectorWeightsLength > 0 && !(vectorWeightsLength == 4 && vectorWeights[0] == 110 && vectorWeights[1] == 117 && vectorWeights[2] == 108 && vectorWeights[3] == 108) {
```
Comment (parkerduckworth) [inline entities/storobj/storage_object.go:172]: Would help readability to break this down into separate conditional functions. This way you can use the naming of the functions to self-document the reasoning behind each of these propositions

Comment (dirkkul) [reply entities/storobj/storage_object.go]: I reformated it and added comments to make it clearer

File: entities/storobj/storage_object.go [hunk 3]
```diff
@@ -113,94 +111,88 @@ func FromBinaryUUIDOnly(data []byte) (*Object, error) {
...
+	// some object members need additional "enrichment". Only do this if necessary, ie if they are actually present
+	if len(props) > 0 ||
+		len(meta) > 0 ||
+		vectorWeightsLength > 0 &&
+			!( // if the length is 4 and the encoded value is "null" (in ascii), vectorweights are not actually present
+			vectorWeightsLength == 4 &&
+				vectorWeights[0] == 110 && // n
+				vectorWeights[1] == 117 && // u
+				vectorWeights[2] == 108 && // l
+				vectorWeights[3] == 108) { // l
```
[the remark this answers is not in the document]
Comment (dirkkul) [reply entities/storobj/storage_object.go:178]: might also work, but I wanted to minimize allocations here

**Pull request merged**